lib/http_negotiate.c

```diff
@@ -148,7 +148,7 @@ CURLcode Curl_output_negotiate(struct connectdata *conn, bool proxy)
     }
     if(!neg_ctx->context) {
       result = Curl_input_negotiate(conn, proxy, "Negotiate");
-      if(result == CURLE_LOGIN_DENIED) {
+      if(result == CURLE_AUTH_ERROR) {
```

Comment:
There are other points in those functions where it returns login denied, should this be CURLE_LOGIN_DENIED || CURLE_AUTH_ERROR ?

Reply:
Thanks for review!  This quirk was originally intended to handle the failure of `gss_init_sec_context()`, which was the common trigger in all the bug reports about unexpected change of behavior caused by 6c6035532383e300c712e4c1cd9fdd749ed5cf59.  I am not sure how the other error paths are common in practise and whether they deserve the same treatment.

Reply:
The other error paths are for real login denied meaning the auth procedure was successful but rejected so there was no actual auth function error, so I think they should continue to return LOGIN_DENIED. What's unclear to me is if LOGIN_DENIED at this point whether it's intended to continue unauthenticated because this is a change in behavior now, since f460370 we let it continue on and I don't know if that is what was intended for actual login denied.

Reply:
It is difficult to guess.  Before 6c6035532383e300c712e4c1cd9fdd749ed5cf59 we did not even try to authenticate until the **server** asked for it.  Now we try to authenticate each time **user** asks for it and try to simulate the old behavior by pretending that nothing happened in case the authentication fails.  I am aware only about bug reports involving the failure of `gss_init_sec_context()`, no idea whether the other failures are worth the same compatibility quirk.  But I have no strong opinion on this and will add `|| result = CURLE_LOGIN_DENIED` if you think it is safer.

Reply:
hm resolving conversation wipes out the comment and doesn't post it, didn't know that...
I looked at the code again and there's no login denied error path here because this is the initial setup only called when no context exists so I think you're ok as is.